File: complaint-system-frontend/src/app/services/assigned-complaint.service.ts
import { Injectable } from '@angular/core';
import { assignedComplaint } from '../model/assignedComplaint.model';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AssignedComplaintService {

  baseUrl:string = "http://localhost:8080/api/assignedComplaint"
  constructor(private http:HttpClient) { }

  addComplaint(assignedComplaint:assignedComplaint){
    return this.http.post(this.baseUrl,assignedComplaint);
  }

  getAllComplaints(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl);
  }

  getAllComplaintsEngineerA(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerA");
  }

  getAllComplaintsEngineerB(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerB");
  }

  getAllComplaintsEngineerC(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerC");
  }

  getAllComplaintsEngineerD(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerD");
  }

  getAllComplaintsEngineerE(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerE");
  }

  getAllComplaintsEngineerF(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerF");
  }

  getAllComplaintsEngineerG(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerG");
  }

  getAllComplaintsEngineerH(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerH");
  }

  getAllComplaintsEngineerI(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerI");
  }

  getAllComplaintsEngineerJ(): Observable<assignedComplaint[]>{
    return this.http.get<assignedComplaint[]>(this.baseUrl+"/getAllCompEngineerJ");
  }
}

File: Complaint-Redressal-System/src/main/java/com/complaint/system/service/AssignedComplaintService.java
package com.complaint.system.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.complaint.system.entity.AssignedComplaint;
import com.complaint.system.repository.AssignedComplaintRepo;

@Service
public class AssignedComplaintService {

	@Autowired
	private AssignedComplaintRepo repo;
	
	public AssignedComplaint addComplaint(AssignedComplaint assignedComplaint) {
		return repo.save(assignedComplaint);
	}
	
	public List<AssignedComplaint> getAllAssignedComplaints(){
		return repo.findAll();
	}

	public void delComplaint(int id) {
		repo.deleteById(id);
	}
	
	public List<AssignedComplaint> getAllCompEngineerA(){
		return repo.getAllComplaintsEngineerA();
	}
	
	public List<AssignedComplaint> getAllCompEngineerB(){
		return repo.getAllComplaintsEngineerB();
	}
	
	public List<AssignedComplaint> getAllCompEngineerC(){
		return repo.getAllComplaintsEngineerC();
	}
	
	public List<AssignedComplaint> getAllCompEngineerD(){
		return repo.getAllComplaintsEngineerD();
	}
	
	public List<AssignedComplaint> getAllCompEngineerE(){
		return repo.getAllComplaintsEngineerE();
	}
	
	public List<AssignedComplaint> getAllCompEngineerF(){
		return repo.getAllComplaintsEngineerF();
	}
	
	public List<AssignedComplaint> getAllCompEngineerG(){
		return repo.getAllComplaintsEngineerG();
	}
	
	public List<AssignedComplaint> getAllCompEngineerH(){
		return repo.getAllComplaintsEngineerH();
	}
	
	public List<AssignedComplaint> getAllCompEngineerI(){
		return repo.getAllComplaintsEngineerI();
	}
	
	public List<AssignedComplaint> getAllCompEngineerJ(){
		return repo.getAllComplaintsEngineerJ();
	}
}
